src/test/java/org/jenkinsci/plugins/workflow/libs/SCMRetrieverTest.java

```diff
@@ -49,6 +51,7 @@
 
     @Url("https://stackoverflow.com/a/49112612/12916")
     @Test public void selfTestLibraries() throws Exception {
+        Assume.assumeFalse("File names are too long for Windows", Functions.isWindows());
```

Comment:
It doesn't feel right to try to nickel and dime the file names to save a few characters ("mp" -> "m", "branch" -> "b", "master" -> "m"), when the main issue (I think) is due to having 2 hashes included from branch-api in the path (One for ci.jenkins.io, and one for the test itself) in addition to the integer suffix on the JenkinsRule's JENKINS_HOME directory. AIUI jenkinsci/branch-api-plugin#129 might make this better.

Reply:
https://github.com/jenkinsci/plugin-pom/pull/125 may help somewhat.

Reply:
3.24 now

Reply:
Still failing, will probably just revert and leave the test ignored for now on Windows. It seems strange that there are nested `junit` folders in the workspace, but I'm not sure if that is related to any of the recent changes: `...\target\tmp\junit5127255811482076289\junit7031506459549069085`